legend-engine-core/legend-engine-core-executionPlan-execution/legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/nodes/helpers/freemarker/FreeMarkerExecutor.java

```diff
@@ -79,17 +86,68 @@ else if (v instanceof StreamingResult)
         return StringUtils.isBlank(templateFunctions) ? process(input, variableMap, templateFunctions) : processRecursively(input, variableMap, templateFunctions);
     }
 
+
+    private static boolean isPlaceHolder(Object object)
+    {
+        return object instanceof String && pattern.matcher((String)object).find();
+    }
+
+    private static class TemplateHashModelOverride implements TemplateHashModel
+    {
+        private Map map;
+        private String templateFunctions;
+
+        TemplateHashModelOverride(Map map, String templateFunctions)
+        {
+            this.map = map;
+            this.templateFunctions = templateFunctions;
+        }
+
+        @Override
+        public TemplateModel get(String s) throws TemplateModelException
+        {
+            Object result = map.get(s);
+            boolean isResultUnchanged = false;
+            while (!isResultUnchanged && isPlaceHolder(result))
+            {
+                String lastProcessedResult = (String) result;
+                result = process((String) result, map, templateFunctions);
+                isResultUnchanged = lastProcessedResult.equals(((String) result).replace("\\\"", "\""));
+            }
+            return objectWrapper.wrap(result);
+        }
+
+        @Override
+        public boolean isEmpty()
+        {
+            return map.isEmpty();
+        }
+    }
+
     public static String processRecursively(String input, Map<String, ?> variableMap, String templateFunctions)
+    {
+        boolean overrideTemplateModelFlag = Boolean.valueOf(System.getProperty("overrideTemplateModel"));
```

Comment:
let's not use a magic string, let's define it somewhere and then refer to it, including in the test.

Reply:
I have set the property key as a class member to FreemarkerExecutor.java, and referencing this for the test cases in TestPlanExecutionForIn.java